src/components/side_panel/select_menu/select_menu.ts

```diff
@@ -19,15 +20,20 @@ export class SelectMenu extends Component<SelectMenuProps, SpreadsheetChildEnv>
   static template = "o-spreadsheet-SelectMenu";
   static components = { Menu };
 
+  menuId = new UuidGenerator().uuidv4();
+
   selectRef = useRef("select");
   selectRect = useAbsoluteBoundingRect(this.selectRef);
 
   state = useState<State>({
     isMenuOpen: false,
   });
 
-  onClick() {
-    this.state.isMenuOpen = true;
+  onClick(ev: MenuMouseEvent) {
+    if (ev.closedMenuId === this.menuId) {
```

Comment:
🤮  Hope we can get rid of this at some point, it's just awful; :'(

Reply:
Probably when we have some kind of Menu/Popover store I hope 😅 But even then I'm not sure the problem really change, if the menu is closed at `capture`, how is the component supposed to know that it shouldn't re-open it when receiving the event ?

Reply:
that is a good question, there may be a way based on "who's the currently opened menu" and 'who's the target"?